database/database_test.go

```diff
@@ -234,3 +236,423 @@ func TestStatesWithStatus(t *testing.T) {
+									Message: "add a line at the end of the page",
+								},
+							},
+						},
+					},
+				},
+			}},
+		},
+	},
+	{
+		"two domains",
+		[]string{"b.test", "c.test"},
+		[]IneligibleDomainState{{Name: "b.test", Policy: "bulk-18-week", Scans: []Scan{ 
+			{
+				ScanTime: time.Unix(1234,54324),
+				Issues: []hstspreload.Issues{
+					{
+						Errors: []hstspreload.Issue{
+							{
+								Code: "formatting error", 
+								Summary: "missing end of page line",
+								Message: "add a line at the end of the page",
+							},
+						},
+					},
+				},
+			},
+		}},
+			{Name: "c.test", Policy: "bulk-18-week", Scans: []Scan{ 
+				{
+					ScanTime: time.Unix(1234,54324),
+					Issues: []hstspreload.Issues{
+						{
+							Errors: []hstspreload.Issue{
+								{
+									Code: "example error", 
+									Summary: "missing example",
+									Message: "add example to code",
+								},
+							},
+						},
+					},
+				},
+			}}},
+	},
+	{
+		"multiple domains",
+		[]string{"e.test", "f.test", "g.test", "h.test", "i.test", "j.test", "k.test"},
+		[]IneligibleDomainState{
+			{Name: "e.test", Policy: "bulk-1-year"},
+			{Name: "f.test", Policy: "bulk-1-year"},
+			{Name: "g.test", Policy: "bulk-18-week"},
+			{Name: "h.test", Policy: "bulk-18-week"},
+			{Name: "i.test", Policy: "bulk-1-year"},
+			{Name: "j.test", Policy: "bulk-18-week"},
+			{Name: "k.test", Policy: "bulk-1-year"},
+		},
+	},
+}
+
+// Test GetIneligibleDomainStates tests getting IneligibleDomainStates from the 
+// database
+func TestGetIneligibleDomainStates(t *testing.T) {
+
+	resetDB()
+
+	// domainStates should be empty as domains are not added to database
+	// test for entry that does not exist
+	domainStates, err := testDB.GetIneligibleDomainStates([]string{"a.test"})
+	if len(domainStates) != 0 {
+		t.Errorf("Empty database should contain no preloaded domains")
+	}
+
+	// add domains to the database
+	var statuses []string
+	statusReport := func(format string, args ...interface{}) {
+		formatted := fmt.Sprintf(format, args...)
+		statuses = append(statuses, formatted)
+	}
+	for _, tt := range getAndDeleteTests {
+
+		err := testDB.SetIneligibleDomainStates(
+			tt.wantStates,
+			statusReport,
+		)
+		if err != nil {
+			t.Errorf("[%s] cannot put states %s", tt.description, err)
+			return
+		}
+	}
+	// get domains from the database
+	for _, tr := range getAndDeleteTests {
+		domainStates, err = testDB.GetIneligibleDomainStates(tr.domainNames)
+		if err != nil {
+			t.Errorf("%s", err)
+		}
+
+		sort.Slice(domainStates, func(i, j int) bool { return domainStates[i].Name < domainStates[j].Name })
+		if len(domainStates) != len(tr.domainNames) {
+			t.Errorf("Incorrect count of states for test %s", tr.description)
+		}
+		for i, domainState := range domainStates {
+			if domainState.Name != (tr.domainNames[i]) {
+				t.Errorf("unexpected domain at position %d for test %s: %#v", i, tr.description, domainState)
+			}
+		}
+	}
```

Comment:
table driven tests should do only one loop over the table. Treat the test like you're writing a test for only one entry in the table.